README.md

```diff
@@ -1,46 +1,160 @@
 # Infomaniak Android Core library
 
-Infomaniak Core library for Android (many purposes)
+Infomaniak Core is a modular Android library used across multiple projects.
+It can be consumed locally using **Gradle Composite Builds**, enabling fast development
+without publishing artifacts, while still supporting the historical **Legacy** module.
 
-## Installation
+---
 
-Import the project as a git modsub-module in the root folder of your project
+## Overview
 
-## Use
+This repository contains:
+
+* The **new modular Core**, designed to be consumed via **composite builds**
+* The **Legacy** module, kept for backward compatibility
+* Shared resources, models, and networking utilities
+
+The composite build integration is powered by the Gradle **Settings plugin**:
+
+```
+com.infomaniak.core.composite.CoreCompositePlugin
+```
+
+📘 For advanced configuration and exact dependency mapping rules,
+**refer to the KDoc of `CoreCompositePlugin`.**
+
+---
+
+## Installation (Composite Build – recommended)
+
+Import the project as a git sub-module in the root folder of your project
+
+To use the **new Core with composite build**, you must update the **host project’s**
+`settings.gradle.kts`.
+
+### 1️⃣ Include Core build logic
+
+```kotlin
+pluginManagement {
+    includeBuild("Core/build-logic")
+}
+```
+
+> ⚠️ Mandatory
+> The Core composite plugin is provided by Core’s build logic.
+
+---
```

Comment:
Do we really need `---` separators between steps of the same category? This feels too heavy. Using # and ## already adds separators automatically, are they not enough?

This could actually apply to the whole readme even